### ISSUE TITLE
Refactor Animal Manure Excretion and Enteric Methane Files 

### ISSUE THREAD

**Issue by tomhuhh** [RuminantFarmSystems/RuFaS#657]
**Is your feature request related to a problem? Please describe.**
The current code is stand-alone code, which can create difficulty down the road.  

**Describe the solution you'd like**
The solution proposed is to add these functions to their respective animal category class. 

**Describe alternatives you've considered**
I am open to alternatives. 

**Additional context**
NA

**Comment by PooyaHekmati** [comment]
TODO: Follow up with author to check if this is still the case.

**Comment by tomhuhh** [comment]
I can confirm these are still stand alone files. We could potentially take these into consideration during the animal module refactoring led by @JoeWaddell. 

Thank you @PooyaHekmati  for following up on these issues!

**Comment by ew3361zh** [comment]
@allisterakun to consult with @tomhuhh and will rewrite this issue to be clearer with a good action plan in place.

Potential solution:
- Convert these standalone methods to helper-functions by moving them to a better place and giving them better names.
- After #1680 is completed, add a router function to the Digestive System class to call these helper functions.

**Comment by allisterakun** [comment]
After a quick review of the current code, here is a proposal for this issue:
    An `AnimalManureExcretions` class will be created in the _**DigestiveSystem**_ submodule to house all methods related to animal manure excretion calculations. This class will have an entry point method, `calculate_manure_excretions()`, that acts as a router. All animals will call on this entry function, which will redirect the call to the corresponding method according to the current life stage of the animal.


for files under the `RUFAS/routines/animal/manure/` folder:

- `calf_manure_excretion.py`
    - this file contains the manure excretion calculation method for _**Calf**_
        - this method will be moved into the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule. All calculations for calves will be routed to this function by `calculate_manure_excretions()`
- `dry_cow_manure_excretion.py`
    - this file contains the manure excretion calculation method for _**Dry Cow**_
        - this method will be moved into the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule. All calculations for dry cows will be routed to this function by `calculate_manure_excretions()`
- `growing_heifer_manure_excretion.py`
    - this file contains the manure excretion calculation method for _**Heifer**_
        - this method will be moved into the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule. All calculations for heifers will be routed to this function by `calculate_manure_excretions()`
- `lactating_cow_manure_excretion.py`
    - this file contains the manure excretion calculation method for _**Lactating Cow**_
        - this method will be moved into the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule. All calculations for lactating cows will be routed to this function by `calculate_manure_excretions()`
    - this file also contains the methane mitigation calculation method; the formula in this method seems to be the same as the ones used in `methane_mitigation.py`
    - Please confirm that methane mitigation only applies to lactating cows
    - Remove this method, and update all calls to this function to use the `MethaneMitigationMethodManager.calculate_methane_mitigation()` method in `methane_mitigation.py`
- `general_manure.py`
    - this file contains a _**TypedDict**_ class `AnimalManureExcretions` that defines the animal manure excretions output, along with two methods for addition and scalar multiplication for `AnimalManureExcretions` objects.
        - This class will be moved into the _**DigestiveSystem**_  submodule and made a stand-alone class. To avoid confusion, it will be renamed `AnimalManureExcretionsOutput`. The two methods will overwrite the addition and multiplication methods for this class.
    - it also has a default animal manure excretions method that returns an `AnimalManureExcretions` with all properties set to 0.0
        - Do we really need this default case?
            - If yes, move this function to the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule.
    - this file also contains the method for phosphorus excretion values calculation (from my understanding, the formula is the same for all animals despite their life stage)
        - Move this function into the `AnimalManureExcretions` class under the _**DigestiveSystem**_ submodule.
- `methane_mitigation.py`
    - Move this class under the _**DigestiveSystem**_ submodule.

@tomhuhh Please let me know what you think!

**Comment by allisterakun** [comment]
@allisterakun will handle most of this issue during the refactoring of the animal classes in issue #1680. He will then come back after the refactoring process and create a new issue for whatever remains to be implemented.

**Comment by allisterakun** [comment]
Closing this issue because it will be taken care of in issue #1927
